src/m_pd.h

```diff
@@ -868,6 +869,7 @@ typedef union
 static inline int PD_BADFLOAT(t_float f)  /* malformed double */
 {
     t_bigorsmall64 pun;
+    if (f == 0) return 0;
```

Comment:
hmm, this turns the highly performance critical `PD_BADFLOAT` into code with branches, whereas previously it was branchless...

a good optimizing compiler might of course generate branchless code from that, but it would be good if that was the case even for simple compilers...

Reply:
that was my first reaction as well but at least it isn't used in very critical areas now afaict

Reply:
indeed, after checking  it seems that in Pd-vanilla it's currently only used in slider/vumeter/`[oscparse]`.

i see that cyclone uses it a bit in its dsp perform routines (but it does so after calling to things like `acosf()` so i think performance is not a concern here :-)).

it would still be great if we had it non-branching.

Reply:
maybe I'm missing some ieee thing but wouldn't it just work to take out the ```(pun.ui == 0)``` in the return?

Reply:
i *think* that is testing for denormals, where the exponent is 0 but the mantissa is non-null, which should be caught.

checking the `pun.ui` value *before* masking should do the trick though.
something like the following untested code:

```C
    t_bigorsmall32 pun;
    unsigned int nonzero;
    pun.f = f;
    nonzero = pun.ui;
    pun.ui &= 0x7f800000;
    return nonzero && ((pun.ui == 0) | (pun.ui == 0x7f800000));
```

Reply:
maybe something like
```
int tmp = pun.ui;
pun.ui &= 0x7f800000;
return (tmp <  0x7f800000) | (pun.ui == 0x7f800000);
```

Reply:
maybe something like
```
tmp = pun.ui
pun.ui &= 0xff800000;
return (tmp <  0x00800000) | (pun.ui >= 0x7f800000);
```

Reply:
ah I always feel like there's some other clever bit puzzle but can't quite figure out how to keep zero out of them

Reply:
Why not simply:
```
t_bigorsmall32 pun;
pun.f = f;
pun.ui &= 0x7f800000;
return (f != 0) & ((pun.ui == 0) | (pun.ui == 0x7f800000));
```

Reply:
Haha, yes, I was just thinking this. This approach will translate way more easily into the 64-bit version.